kratos/utilities/parallel_utilities.cpp

```diff
@@ -30,6 +30,18 @@ namespace {
     std::once_flag flag_once;
 }
 
+int ParallelUtilities::GetThreadId()
+{
+#ifdef KRATOS_SMP_OPENMP
+    const int ithreads = omp_get_thread_num();
+    return ithreads;
+#elif defined(KRATOS_SMP_CXX11)
+    KRATOS_ERROR << "No equivalent definition of GetThreadId in C++11. Replace current implementation to something that fits the C++ STL requirements" << std::endl;
```

Comment:
I would have done this long ago, but there is no (easy) equivalent for this with Cxx

@RiccardoRossi and I discussed this a lot

#9985 will be the proper solution for this, not requiring the threadId at all

Reply:
True. In any case, right now current code that relies in OpenMPUtils will not work with SMP_CXX11, the code is more like a transition to remove OpenMPUtils. I can just simply revert change and wait until #9985 is merged

Reply:
Separated in https://github.com/KratosMultiphysics/Kratos/pull/11116